pandas/plotting/_matplotlib/core.py

```diff
@@ -579,13 +579,24 @@ def legend_title(self) -> Optional[str]:
             stringified = map(pprint_thing, self.data.columns.names)
             return ",".join(stringified)
 
-    def _add_legend_handle(self, handle, label, index=None):
-        if label is not None:
-            if self.mark_right and index is not None:
-                if self.on_right(index):
-                    label = label + " (right)"
-            self.legend_handles.append(handle)
-            self.legend_labels.append(label)
+    def _mark_right_label(self, label: str, index: int) -> str:
+        """
+        Append ``(right)`` to the label of a line if it's plotted on the right axis.
+
+        Note that ``(right)`` is only appended when ``subplots=False``.
+        """
+        if not self.subplots and self.mark_right and self.on_right(index):
+            label += " (right)"
+        return label
+
+    def _add_legend_handle(self, handle: Artist, label: str) -> None:
```

Comment:
nitpick on the naming since we are refactoring here, maybe rephrase to `_add_legend_handle_label` or `_append_legend_handles_labels`?

Reply:
good suggestion, thanks!